sdks/java/core/src/main/java/org/apache/beam/sdk/util/state/InMemoryTimerInternals.java

```diff
@@ -97,6 +98,18 @@ public Instant getNextTimer(TimeDomain domain) {
   }
 
   @Override
+  public void setTimer(StateNamespace namespace, String timerId, Instant target,
+      TimeDomain timeDomain) {
+    WindowTracing.trace(
+        "TestTimerInternals.setTimer: {}/{} {} {}", namespace, timerId, target, timeDomain);
+    // TODO: actually track timers by ID
+    TimerData timer = TimerData.of(namespace, target, timeDomain);
+    if (existingTimers.add(timer)) {
```

Comment:
Are you avoiding the call to `setTimer(TimerData)` here because you want to keep a different trace ?